### PR TITLE
sandbox-classic: Remove default parameters in `SqlLedger.Owner`.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
The defaults can be dangerous.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by kamil-da — APPROVED** [review]

File: ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala [hunk 1]
```diff
@@ -93,6 +93,9 @@ private[sandbox] object LedgerResource {
           metrics = new Metrics(metrics),
           lfValueTranslationCache = LfValueTranslationCache.Cache.none,
           engine = new Engine(),
+          validatePartyAllocation = false,
+          enableAppendOnlySchema = true,
```
Comment (rautenrieth-da) [inline ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala:97]: This is changing the behavior, the test will be running append-only schema now. Was this intended?

Comment (rautenrieth-da) [reply ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala:97]: Changing all tests to use the append-only schema is something that we should do now, but I'd rather have this in a separate PR.

**Review by rautenrieth-da — CHANGES_REQUESTED** [review]

File: ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala (hunk 1 above)
Comment (kamil-da) [reply ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala:97]: Good point @rautenrieth-da. 
I thought it's intended to change to the append-only but I agree that it'd be better to do this in a separate PR.

Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala:97]: Oh, good point. Thanks!

Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/LedgerResource.scala]: Reverted to `false`.

**Review by rautenrieth-da — APPROVED** [review]

**Pull request merged**